src/main/app/document/DocumentManagementController.ts

```diff
@@ -224,34 +459,61 @@ export class DocumentManagerController extends PostController<AnyObject> {
-    const documentsUploadedKey = 'applicantDocumentsUploaded';
-    const documentsUploaded = (req.session.userCase[documentsUploadedKey] as ListValue<UploadDocumentList>[]) ?? [];
-
-    const documentIndexToDelete = parseInt(req.params.index, 10);
-    const documentToDelete = documentsUploaded[documentIndexToDelete];
-    if (!documentToDelete?.value?.value?.citizenDocument.document_url) {
-      return res.redirect(UPLOAD_DOCUMENT);
+  public async deleteDocument(req: AppRequest<Partial<CaseWithId>>, res: Response): Promise<void> {
+    const isApplicant = req.query.isApplicant;
+    let redirectUrl;
+    if (!req?.session?.userCase) {
+      const initData = { id: '1661509886231065', state: State.successAuthentication, serviceType: '' };
+      req.session.userCase = initData;
     }
-    const documentUrlToDelete = documentToDelete?.value?.value?.citizenDocument.document_url;
-
-    // documentsUploaded[documentIndexToDelete].value = null;
-
-    req.session.userCase = await req.locals.api.triggerEvent(
-      req.session.userCase.id,
-      { [documentsUploadedKey]: documentsUploaded },
-      CITIZEN_UPDATE
-    );
-
-    const documentManagementClient = this.getDocumentManagementClient(req.session.user);
-    await documentManagementClient.delete({ url: documentUrlToDelete });
-
-    req.session.save(err => {
-      if (err) {
-        throw err;
+    const caseworkerUser = await getSystemUser();
+    //req.session.userCase['applicantUploadFiles'];
+    const documentIdToDelete = req.params.documentId;
+    const deleteDocumentDetails = {
+      caseId: req.session.userCase.id,
+      documentId: documentIdToDelete,
+    };
+    const deleteDocumentRequest = new DeleteDocumentRequest(deleteDocumentDetails);
+    const client = new CosApiClient(caseworkerUser.accessToken, 'http://localhost:3001');
+    const deleteCitizenDocFromCos = await client.deleteCitizenStatementDocument(caseworkerUser, deleteDocumentRequest);
+    if (deleteCitizenDocFromCos === 'SUCCESS') {
+      if (isApplicant === 'Yes') {
+        req.session.userCase.applicantUploadFiles?.forEach((document, index) => {
+          if (document.id === documentIdToDelete) {
+            req.session.userCase.applicantUploadFiles?.splice(index, 1);
+          }
+        });
+      } else {
+        req.session.userCase.respondentUploadFiles?.forEach((document, index) => {
+          if (document.id === documentIdToDelete) {
+            req.session.userCase.respondentUploadFiles?.splice(index, 1);
+          }
+        });
       }
-      return res.redirect(UPLOAD_DOCUMENT);
-    });
+      const caseDataFromCos = await client.retrieveByCaseId(req.session.userCase.id, caseworkerUser);
+      req.session.userCase.citizenUploadedDocumentList = caseDataFromCos.citizenUploadedDocumentList;
+      req.session.errors = [];
+    } else {
+      req.session.errors?.push({ errorType: 'Document could not be deleted', propertyName: 'uploadFiles' });
+    }
+    if (isApplicant === 'Yes') {
```

Comment:
Seen this logic at multiple places, see if we can put it in one common method.

Reply:
its a different logic used at two places